davdwhyte87/iReporter
# #162575273 Restructure api

**What does this pR do?**
This pull request adds some structure to the api and documentation to the code base

**Description of task to be completed?**
- configure eslint for js Doc
- correct code conventions
- code documentation
- classes as controllers in stead of methods
- remove validation errors from controllers 
- write new tests to ensure the api is not broken

**How can this be manually tested?**
- clone the repo
- cd into the project directory
- run npm i to install dependencies
- install postgreSQL and reate database ireporter 
- check example.env and add .env file, then fill in the details
- run npm start to start the server
- access the endpoints from `localhost:$POPRT`


**Relevant pivotal tracker stories?**
`162575273`
